src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -370,14 +378,19 @@ void UDPEndPointImplLwIP::LwIPReceiveUDPMessage(void * arg, struct udp_pcb * pcb
                                                 u16_t port)
 {
     UDPEndPointImplLwIP * ep = static_cast<UDPEndPointImplLwIP *>(arg);
+
+    LockChipStack();
```

Comment:
Why are these new locks here?  Especially when they are no-ops?

Reply:
@bzbarsky-apple please see https://github.com/project-chip/connectedhomeip/issues/28590#issuecomment-1705582187 (sorry, i should have commented here). TLDR is: i think we should, but at the moment we can't. i'll need guidance here on how to proceed.

Reply:
@rojer This method was designed to be able to run from outside the Matter thread, without locking, because that's how LwIP calls it.  If it's not doing that correctly, it should be fixed, instead of adding locks...

Reply:
yes, LwIPReceiveUDPMessage will be invoked on another thread and since it pokes at the endpoint's internals, the endpoint's reference counter needs to be protected. i see that packet filter is also being invoked, so it should be protected too.
but how to do that other than by adding a lock (or locking the stack, as i attempted to do)? i'm confused.

Reply:
Packet filter is expected to be called from random threads; it documents that explicitly in its API documentation.  So the synchronization to be done there, if any, is the packet filter's responsibility.

The refcount of the endpoint is not touched in LwIPReceiveUDPMessage.

`mDelayReleaseCount` is an `std::atomic` precisely to avoid worrying about races on it.  That said, we do need to make sure we can't be inside LwIPReceiveUDPMessage on a different thread at the time when the `mDelayReleaseCount` check in `UDPEndPointImplLwIP::CloseImpl` happens.  I don't know whether the earlier completion of the `udp_remove` call there guarantees that, but I would hope so...  If it does not, we need to do something to synchronize here.

Reply:
ok, thanks for the explanation! i think we're good: `UDPEndPointImplLwIP::CloseImpl` runs `udp_remove` on the TCPIP task (or with core lock), after that no more receive callbacks will be delivered for that PCB. hence, we don't need additional synchrnization after all, so i removed CHIP stack locking calls.